src/components/Buttons/ToggleChatButton/ToggleChatButton.test.tsx

```diff
@@ -0,0 +1,28 @@
+import React from 'react';
+import { shallow } from 'enzyme';
+import { Button } from '@material-ui/core';
+
+import ChatIcon from '../../../icons/ChatIcon';
+import ToggleChatButton from './ToggleChatButton';
+
+import useChatContext from '../../../hooks/useChatContext/useChatContext';
+
+jest.mock('../../../hooks/useChatContext/useChatContext');
+const mockUseChatContext = useChatContext as jest.Mock<any>;
+
+const mockToggleChatWindow = jest.fn();
+mockUseChatContext.mockImplementation(() => ({ setIsChatWindowOpen: mockToggleChatWindow }));
+
+describe('the ToggleChatButton component', () => {
+  it('should render correctly when chat is enabled', () => {
+    const wrapper = shallow(<ToggleChatButton />);
+    expect(wrapper.prop('startIcon')).toEqual(<ChatIcon />);
+    expect(wrapper.text()).toBe('Chat');
+  });
+
+  it('should call the correct toggle function when clicked', () => {
+    const wrapper = shallow(<ToggleChatButton />);
+    wrapper.find(Button).simulate('click');
+    expect(mockToggleChatWindow).toHaveBeenCalled();
```

Comment:
I think here we can test that the button is correctly negating the value of `isChatWindowOpen`. Let's define `isChatWindowOpen` (say as `false`) on line 14, then here on line 26 we can use `.toHaveBeenCalledWith(true)`.